Earthfile

```diff
@@ -0,0 +1,94 @@
+# Earthfile
+VERSION 0.6
+
+ARG PYTHON_MAJOR=3.9
+ARG PYTHON_VERSION=$PYTHON_MAJOR.12
+FROM python:$PYTHON_VERSION-alpine
+
+RUN apk add --no-cache git gcc libffi-dev build-base bash
+ARG SOURCE_DIR="superai"
+ENV AWS_DEFAULT_REGION="us-east-1"
+# Flag to enable codeartifact login
+ARG INTERNAL=true
+
+pre-commit:
+    RUN pip install pre-commit==2.17.0
+    COPY . ./app
+    WORKDIR /app
+    RUN --mount=type=cache,target=/root/.cache/pre-commit pre-commit run --all-files
+
+semgrep:
+    ARG SEMGREP_VERSION=0.86.5
+    RUN pip install semgrep==$SEMGREP_VERSION
+    # Download rules separately for caching purposes in .ci/semgrep_rules
+    RUN mkdir /rules
+    COPY . ./app
+    WORKDIR /app
+    RUN semgrep --metrics=off --error --severity ERROR --disable-version-check --config .ci/semgrep_rules
+
+linter:
+    BUILD +semgrep
+    BUILD +pre-commit
+
+PIP_INSTALL:
+    COMMAND
+    ARG REQTARGET="."
+    IF  [ "$INTERNAL" = "true" ]
+        RUN --mount=type=secret,id=+secrets/aws,target=/root/.aws/credentials \
+            --mount=type=cache,target=/root/.cache/pip \
+            aws codeartifact login --tool pip --domain superai --repository pypi-superai-internal && \
```

Comment:
We shouldn't require the superai-internal repository, we should use the pypi-superai

Reply:
I checked the idea behind this. This is handled internally in this file here https://github.com/mysuperai/superai-sdk-automation/blob/main/vars/buildPipeline.groovy. The logic differs for internal and external SDK

Reply:
If you do `aws codeartifact login --tool pip --domain superai --repository pypi-superai` should work for externals. But please test it first